### PR TITLE
feat: use `DefaultAzureCredential` in Azure Key Vault

### PR DESCRIPTION
**Pull request by bscholtes1A**
## What this PR changes/adds

Use `DefaultAzureCredential` in Azure Key Vault extension, which removes the EDC code logic required to select the proper `TokenCredential` to be used for authenticating to the Vault. Also it now enables password-less authentication to the Vault from other managed Azure services.

## Why it does that

Use `TokenCredential` selection logic from Azure library instead of duplicating it. Similar to what is done in https://github.com/eclipse-edc/Technology-Azure/pull/77.

## Further notes

Note that projects using this extension will have to update the environment variables (Catena-X guys @paullatzelsperger @wolf4ood  @ndr-brt this will translate into an update in the Tractus-X Helm charts):
 - `EDC_VAULT_CLIENTID` must be replaced by `AZURE_CLIENT_ID`
 - `EDC_VAULT_TENANTID` must be replaced by `AZURE_TENANT_ID`
 - `EDC_VAULT_CLIENTSECRET` must be replaced by `AZURE_CLIENT_SECRET`
 - `EDC_VAULT_CERTIFICATE` must be replaced by `AZURE_CLIENT_CERTIFICATE_PATH`

Please find [here ](https://learn.microsoft.com/fr-fr/dotnet/api/azure.identity.environmentcredential?view=azure-dotnet) the Azure doc which specify these environment variables.

### CODE REVIEW

File: extensions/common/vault/vault-azure/src/test/java/org/eclipse/edc/vault/azure/AzureVaultExtensionTest.java [hunk 1]
```diff
@@ -8,93 +8,59 @@
...
-        when(context.getSetting("edc.vault.tenantid", null)).thenReturn(TENANT_ID);
-        when(context.getMonitor()).thenReturn(monitor);
-    }
-
-    @Test
-    void neitherSecretOrCertificateProvided_shouldThrowException() {
-        assertThrows(AzureVaultException.class, () -> extension.initialize(context));
+    void setUp(ServiceExtensionContext context, ObjectFactory factory) {
+        this.context = context;
+        extension = factory.constructInstance(AzureVaultExtension.class);
```
Comment (ndr-brt) [inline extensions/common/vault/vault-azure/src/test/java/org/eclipse/edc/vault/azure/AzureVaultExtensionTest.java:45]: no need to explicitly instantiate the extension, you can inject the `AzureVaultExtension` directly in the tests, same for the context

**Review by ndr-brt — COMMENTED** [review]
> Note that projects using this extension will have to update the environment variables (Catena-X guys @paullatzelsperger @wolf4ood  @ndr-brt this will translate into an update in the Tractus-X Helm charts):
> - `EDC_VAULT_CLIENTID` must be replaced by `AZURE_CLIENT_ID`
> - `EDC_VAULT_TENANTID` must be replaced by `AZURE_TENANT_ID`
> - `EDC_VAULT_CLIENTSECRET` must be replaced by `AZURE_CLIENT_SECRET`
> - `EDC_VAULT_CERTIFICATE` must be replaced by `AZURE_CLIENT_CERTIFICATE_PATH`

maybe it could worth explaining the azure environment variables that should be used (they aren't mentioned anywhere in the code)

File: extensions/common/vault/vault-azure/src/test/java/org/eclipse/edc/vault/azure/AzureVaultExtensionTest.java (hunk 1 above)
Comment (bscholtes1A) [reply extensions/common/vault/vault-azure/src/test/java/org/eclipse/edc/vault/azure/AzureVaultExtensionTest.java:45]: They will depend on the method you choose to authenticate, as described in the Azure spec. For some, a secret is needed, for other it isn't.
I can reference the Azure doc in the doc tough, is it ok?

Comment (ndr-brt) [reply extensions/common/vault/vault-azure/src/test/java/org/eclipse/edc/vault/azure/AzureVaultExtensionTest.java:45]: sure it is

Comment (bscholtes1A) [reply extensions/common/vault/vault-azure/src/test/java/org/eclipse/edc/vault/azure/AzureVaultExtensionTest.java:45]: I just added a readme file

**Review by ndr-brt — APPROVED** [review]
LGTM

**Review by wolf4ood — APPROVED** [review]
LGTM

**Pull request merged**